### PR TITLE
re-use tags routines for folders

### PR DESCRIPTION
**Pull request by AlicVB**
remove and simplify the code for folders tree, as it works exactly like tags, except the separator and some minor things.
it should be 1:1 for features.

### CODE REVIEW

File: src/libs/collect.c [hunk 1]
```diff
@@ -86,8 +84,8 @@ typedef enum dt_lib_collect_cols_t
   DT_LIB_COLLECT_COL_ID,
   DT_LIB_COLLECT_COL_TOOLTIP,
   DT_LIB_COLLECT_COL_PATH,
-  DT_LIB_COLLECT_COL_COUNT,
   DT_LIB_COLLECT_COL_VISIBLE,
+  DT_LIB_COLLECT_COL_NOTHERE,
```
Comment (LebedevRI) [inline src/libs/collect.c]: It means that the file is missing from the current file system?
Not sure about the name, it took me some time to parse

**Comment by LebedevRI** [comment]
Ok, looks good, only one thing:
Let's image that db only has images from `/home/lebedevri/*`
Now, when "folders" are selected, `lebedevri` will be the first-level folder.
And in this pr, `home` will be the first-level folder...

**Comment by AlicVB** [comment]
Replaced NOTHERE by UNREACHABLE. I hope this is more meaningful...
Fix the root for folders

**Comment by LebedevRI** [comment]
```
[14:28:27] <houz> maybe write down a small spec in a few lines that describes how it _should_ behave? so we have something simple to discuss/agree on and later test against?
[14:29:19] <LebedevRI> in library you have /foo/bar/12.jpg /foo/bar/baz/34.CR2
[14:29:41] <LebedevRI> git master shows it as: bar/12.jpg bar/baz/34.CR2
[14:29:58] <LebedevRI> #1112 initially showed it as /foo/bar/12.jpg /foo/bar/baz/34.CR2
[14:30:19] <LebedevRI> now #1112 shows it as 12.jpg baz/34.CR2
[14:30:35] <LebedevRI> ^ which behavior do we want?
[14:31:04] <houz> i somehow agree that you want bar/so you have a single point to collapse, while not having foo/ because that would just be an empty level that might be annoying
[14:31:28] <LebedevRI> that is what i'm saying too
[14:32:05] <houz> LebedevRI: then we should write it down, so the next person rewriting collect in a few years can just read it up
```

File: src/libs/collect.c [hunk 2]
```diff
@@ -765,92 +562,136 @@ static void set_properties(dt_lib_collect_rule_t *dr)
...
-      refilter(model, dr);
-      expand_tree(d->treeview_folders, dr);
+      if(level > 0)
+      {
+        sqlite3_stmt *stmt = NULL;
+        gchar *pth, *query = NULL;
+        int id = -1;
+        // Check if this path also matches a filmroll
+        gtk_tree_model_get(model, &iter, DT_LIB_COLLECT_COL_PATH, &pth, -1);
+        query = dt_util_dstrcat(query, "select id from film_rolls where folder like '%s'", pth);
```
Comment (LebedevRI) [inline src/libs/collect.c]: DT_DEBUG_SQLITE3_PREPARE_V2() + DT_DEBUG_SQLITE3_BIND_INT() is better

**Pull request merged**